### Capstone Project -  Perfumes Recommendation System 
 

#### Import Packages

In [368]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt

#### Read Dataset:
- Perfumes Dataset: 
This dataset contain Perfumes data: Perfumes_Name, Brand, Describtion, Price, Gender, Product_type, Fragrance_Family, Character, Average Rate, Ingredients, Top, Middle and Base Notes, image_URL.

- Reviewa Dataset: 
This dataset contain User nickname, Perfume_name, Brand, Fragrance_Family, Overall_rating, User_rating, Revirews, Date.   

In [369]:
Perfumes_df = pd.read_csv("C:/Users/hmq4/OneDrive/Desktop/Data Science- Misk/chrome/Misk_DSI_Capstone_Project/Dataset/Datasets/Perfume_Dataset.csv")
Reviews_df = pd.read_csv("C:/Users/hmq4/OneDrive/Desktop/Data Science- Misk/chrome/Misk_DSI_Capstone_Project/Dataset/Datasets/Reviews_Dataset.csv")

### 1. Perfumes Dataset Cleaning 

In [370]:
Perfumes_df.head()

Name    Price  \
0         Dolce & Gabanna L'imperatrice 3 Pour Femme  199 SAR   
1                           Roberto Cavalli Paradiso  169 SAR   
2                           Yves Saint Laurent Libre  383 SAR   
3                                Mancera Red Tobacco  499 SAR   
4  Giorgio Armani Emporio Armani Stronger With Yo...  375 SAR   

                                         Description  Rate Rating_count  \
0  Perfume for the energetic woman who is a hero ...     5     6 Rating   
1  Woody floral fragrance, a subtle aroma that ma...  4.95    17 Rating   
2  This perfume is a reflection of Freedom, speci...     5     3 Rating   
3  Mancera Red Tobacco is the new oriental, woody...  4.38     8 Rating   
4  for every romantic gentle man, This strong fas...     5     3 Rating   

                                             Details  \
0  ['Dolce&Gabbana', 'Women', 'Perfume', 'Romanti...   
1  ['Roberto Cavalli', 'Women', 'Perfume', 'Chari...   
2  ['Yves Saint Laurent', 'Women', 'Perfume', 'Ro...   
3  ['Mancera', 'Unisex', 'Perfume', 'Charismatic'...   
4  ['Giorgio Armani', 'Men', 'Perfume', 'Charisma...   

                                               image               Brand  \
0  https://assets.goldenscent.com/catalog/product...       Dolce&Gabbana   
1  https://assets.goldenscent.com/catalog/product...     Roberto Cavalli   
2  https://assets.goldenscent.com/catalog/product...  Yves Saint Laurent   
3  https://assets.goldenscent.com/catalog/product...             Mancera   
4  https://assets.goldenscent.com/catalog/product...      Giorgio Armani   

   Gender Product_Type    Character Fragrance_Family    Size  Year  \
0   Women      Perfume     Romantic           Floral  100 ml  2009   
1   Women      Perfume  Charismatic            Woody   50 ml  2015   
2   Women      Perfume     Romantic           Floral   90 ml  2019   
3  Unisex      Perfume  Charismatic         Oriental  120 ml  2017   
4     Men      Perfume  Charismatic         Aromatic  100 ml  2019   

                                         Ingredients    Concentration  \
0  Watermerlon, Kiwi, Pink Cyclamen, Musk, Pink P...  Eau de Toilette   
1  Citrus, mandarin, bergamot, jasmine, pine, cyp...    Eau de Parfum   
2  Mandarin Orange, lavendar, black currant, peti...    Eau de Parfum   
3  Saffron, Cinnamon, Incense, Nutmeg, White Peac...    Eau de Parfum   
4  Spices, Violet, Lavender, Sweet Toffee, Carame...    Eau de Parfum   

                                            Top_note  \
0                         Pink pepper, kiwi, rhubarb   
1                     Citruses , Mandarin , Bergamot   
2  Mandarin Orange, Lavendar, Black Currant, Peti...   
3  Saffron, Cinnamon, Incense, Nutmeg, White peac...   
4                  Pink pepper, Juniper, Violet leaf   

                        Middle_note  \
0     jasmine, cyclamen, watermelon   
1                           Jasmine   
2           Jasmine, Orange Blossom   
3                Patchouli, Jasmine   
4  Lavender, Sage, Toffee, Cinnamon   

                                           Base_note  
0                     musk, sandalwood, lemon trees.  
1                 Cypress, Parasol pine, Pink laurel  
2                    Vanilla, Cedar, Musk, Ambergris  
3  Tobacco, Amber, Woody notes, Vetiver, Vanilla,...  
4                  Tonka bean, Suede, Amber, Vanilla

In [371]:
Perfumes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9512 entries, 0 to 9511
Data columns (total 19 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Name              9512 non-null   object
 1   Price             9512 non-null   object
 2   Description       9467 non-null   object
 3   Rate              9512 non-null   object
 4   Rating_count      9505 non-null   object
 5   Details           9512 non-null   object
 6   image             9512 non-null   object
 7   Brand             9178 non-null   object
 8   Gender            9156 non-null   object
 9   Product_Type      9011 non-null   object
 10  Character         8694 non-null   object
 11  Fragrance_Family  8284 non-null   object
 12  Size              8001 non-null   object
 13  Year              7662 non-null   object
 14  Ingredients       7506 non-null   object
 15  Concentration     6897 non-null   object
 16  Top_note          6134 non-null   object
 17  Middle_note   

In [372]:
print(f'This dataset contains: {Perfumes_df.shape}')

This dataset contains: (9512, 19)


In [373]:
# I will drop Details column because it only contain lists of all the details which I have split them into columns  
Perfumes_df.drop("Details",  axis='columns', inplace=True)

In [374]:
# After droping we left with 18 columns
Perfumes_df.shape

(9512, 18)

In [375]:
# Removing Duplicate rows 
Perfumes_df.duplicated().sum()

2926

In [695]:
Perfumes_df.drop_duplicates(inplace=True)

In [696]:
#DataFrame size after droping duplicates
Perfumes_df.shape

(6566, 18)

### Cleaning Dataset

- Brand Column 

In [378]:
print(f'This dataset contain: {Perfumes_df.Brand.nunique()}')

This dataset contain: 463


- Price Column 

In [379]:
if (Perfumes_df.Price == "none").any():
    Perfumes_df.Price.replace("none", 0, inplace=True)

In [380]:
Perfumes_df['Price'] = Perfumes_df.Price.str.split(' ').str[0]

In [381]:
Perfumes_df.Price = pd.to_numeric(Perfumes_df.Price)

In [382]:
print(Perfumes_df.Price.min())
print(Perfumes_df.Price.max())


6.0
8193.0


- Description Column 

In [383]:
Perfumes_df.Description.isnull().sum()

45

- Size Column 

In [384]:
# some columns have unrelevant data like size
Perfumes_df.Size.value_counts()

100 ml                                            2149
50 ml                                              882
75 ml                                              319
55 ml                                              149
125 ml                                             148
                                                  ... 
Peony Flower                                         1
Flowers, Jasmine                                     1
Apple, Cinnamon, Saffron, Sandalwood, Agarwood       1
Cambodian Oud Oil                                    1
Black tea, clary sage, guaiac wood                   1
Name: Size, Length: 239, dtype: int64

In [385]:
Perfumes_df[Perfumes_df.Size == "Floral"]

Name  Price  \
143         Narciso Rodriguez For Her Hair Mist - 30 ml  207.0   
435         Narciso Rodriguez Narciso Hair Mist - 30 ml  246.0   
639                Carolina Herrera Good Girl Hair Mist  225.0   
864   Paris Hilton Heiress Gift Set - EDP 100 ml + B...  295.0   
1063  Givenchy L'interdit Set (EDT 50 ml + Body Loti...  369.0   
...                                                 ...    ...   
8917      Exuma Parfums Body Mist Profumo Rosa - 100 ml  276.0   
8946          Al Dakheel Oud Dream 3 Body Mist - 100 ml   46.0   
8954            Al Dakheel Oud Crown Body Mist - 100 ml   46.0   
9184        Narciso Rodriguez For Her Hair Mist - 30 ml  207.0   
9476        Narciso Rodriguez Narciso Hair Mist - 30 ml  246.0   

                                            Description  Rate Rating_count  \
143   Narciso Rodriguez Hair Mist for Her has a Flor...  none         none   
435   Narciso Rodriguez Narciso Hair Mist A light an...  none         none   
639   Carolina Herrera Good Girl Hair Mist for Women...     5     1 Rating   
864   Paris Hilton Heiress Gift Set consists of an E...  none         none   
1063  This set from Givenchy includes a 50 ml EDT an...  none         none   
...                                                 ...   ...          ...   
8917  A perfect body mist full of oriental scents an...  none         none   
8946  A wonderful body mist enriched with an enchant...  none         none   
8954  An amazing body mist with a charming and refre...  none         none   
9184  Narciso Rodriguez Hair Mist for Her has a Flor...  none         none   
9476  Narciso Rodriguez Narciso Hair Mist A light an...  none         none   

                                                  image            Brand  \
143   https://assets.goldenscent.com/catalog/product...            30 ml   
435   https://assets.goldenscent.com/catalog/product...            30 ml   
639   https://assets.goldenscent.com/catalog/product...            30 ml   
864   https://assets.goldenscent.com/catalog/product...    Eau de Parfum   
1063  https://assets.goldenscent.com/catalog/product...  Eau de Toilette   
...                                                 ...              ...   
8917                                               none           100 ml   
8946                                               none           100 ml   
8954                                               none           100 ml   
9184  https://assets.goldenscent.com/catalog/product...            30 ml   
9476  https://assets.goldenscent.com/catalog/product...            30 ml   

                 Gender Product_Type    Character Fragrance_Family    Size  \
143   Narciso Rodriguez        Women    Hair Mist        Glamorous  Floral   
435   Narciso Rodriguez        Women    Hair Mist        Glamorous  Floral   
639    Carolina Herrera        Women    Hair Mist         Feminine  Floral   
864        Paris Hilton        Women  Perfume Set        Glamorous  Floral   
1063           Givenchy        Women  Perfume Set    Sophisticated  Floral   
...                 ...          ...          ...              ...     ...   
8917      Exuma Parfums        Women    Body Mist          Sensual  Floral   
8946     Al Dakheel Oud       Unisex    Body Mist          Sensual  Floral   
8954     Al Dakheel Oud       Unisex    Body Mist          Sensual  Floral   
9184  Narciso Rodriguez        Women    Hair Mist        Glamorous  Floral   
9476  Narciso Rodriguez        Women    Hair Mist        Glamorous  Floral   

                                      Year  \
143              Rose, Iris, Powdery, Musk   
435   Rose, Gardenia, Musk, Cedar, Vetiver   
639                                   2017   
864                                   2006   
1063                                  2018   
...                                    ...   
8917                                  2018   
8946                                   NaN   
8954                    Tangerine, Jasmine   
9184       

In [386]:
# replacing Floral with the size Stored in Brand columns 
Perfumes_df['Size'] = np.where(Perfumes_df['Size'] == "Floral", Perfumes_df['Brand'], Perfumes_df['Size'])

In [387]:
Perfumes_df.Size.value_counts()

100 ml                                            2156
50 ml                                              883
75 ml                                              319
55 ml                                              149
125 ml                                             148
                                                  ... 
Sage and Saffron                                     1
Peony Flower                                         1
Flowers, Jasmine                                     1
Apple, Cinnamon, Saffron, Sandalwood, Agarwood       1
Black tea, clary sage, guaiac wood                   1
Name: Size, Length: 240, dtype: int64

In [388]:
#Perfumes_df[Perfumes_df.Size == "Tunisian Neroli Oil,Bulgarian Rose, Turkish Rose,Indonesian Patchouli"]
if (Perfumes_df.Size == "Tunisian Neroli Oil,Bulgarian Rose, Turkish Rose,Indonesian Patchouli").any():
    Perfumes_df.Fragrance_Family.replace("100 ml", "NaN", inplace=True)
    Perfumes_df.Size.replace("Tunisian Neroli Oil,Bulgarian Rose, Turkish Rose,Indonesian Patchouli", "100 ml", inplace=True)
    Perfumes_df.Year.replace("Eau de Toilette", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Tunisian Neroli Oil", "Tunisian Neroli Oil,Bulgarian Rose, Turkish Rose,Indonesian Patchouli", inplace=True)
    Perfumes_df.Concentration.replace("Bulgarian Rose, Turkish Rose", "Eau de Toilette", inplace=True)
    Perfumes_df.Top_note.replace("Indonesian Patchouli", "Tunisian Neroli Oil", inplace=True)
    Perfumes_df.Middle_note.fillna("Bulgarian Rose, Turkish Rose", inplace=True)
    Perfumes_df.Base_note.fillna("Indonesian Patchouli", inplace=True)

In [389]:
Perfumes_df.Size.value_counts()

100 ml                                            2157
50 ml                                              883
75 ml                                              319
55 ml                                              149
125 ml                                             148
                                                  ... 
Orange, Saffron, Caramel                             1
Bergamot, Pear, Pink peppers, Maryam's incense       1
Grapefruit ,Guava                                    1
Sage and Saffron                                     1
Black tea, clary sage, guaiac wood                   1
Name: Size, Length: 239, dtype: int64

In [390]:
if (Perfumes_df.Size == "Eau Fraiche").any():
    Perfumes_df.Brand.replace("Eau Fraiche", "Chanel", inplace=True)
    Perfumes_df.Gender.replace("Chanel", "Women", inplace=True)
    Perfumes_df.Product_Type.replace("Women","Perfume Set", inplace=True)
    Perfumes_df.Character.replace("Perfume Set", "Sensual", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Sensual", "Floral", inplace=True)
    Perfumes_df.Size.replace("Eau Fraiche", "NaN", inplace=True)
    Perfumes_df.Year.replace("Citron, jasmine, teak", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("-", "Citron, jasmine, teak", inplace=True)
    Perfumes_df.Concentration.replace("-", "NaN", inplace=True)
    Perfumes_df.Middle_note.replace("Bulgarian Rose, Turkish Rose", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Indonesian Patchouli", "NaN", inplace=True)


#Perfumes_df['Size'].replace("Eau Fraiche", "NaN", inplace=True)

In [391]:
if (Perfumes_df.Size == "citron,teak wood, jasmine.").any():
    Perfumes_df.Size.replace("citron,teak wood, jasmine.", "35 ml", inplace=True)
    Perfumes_df.Year.replace("citron", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("teak wood", "citron,teak wood, jasmine", inplace=True)
    Perfumes_df.Top_note.replace("35 ml", "citron", inplace=True)
    Perfumes_df.Concentration.replace("jasmine", "NaN", inplace=True)
    Perfumes_df.Middle_note.fillna("teak wood", inplace=True)
    Perfumes_df.Base_note.fillna("jasmine", inplace=True)


In [392]:
if (Perfumes_df.Size == "grapefruit, quince, jasmine, white musk.").any():
    Perfumes_df.Size.replace("grapefruit, quince, jasmine, white musk.", "NaN", inplace=True)
    Perfumes_df.Year.replace("grapefruit", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("quince, jasmine", "grapefruit, quince, jasmine, white musk", inplace=True)
    Perfumes_df.Concentration.replace("white musk.", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("citron","grapefruit", inplace=True)
    Perfumes_df.Middle_note.fillna("quince, jasmine", inplace=True)
    Perfumes_df.Base_note.fillna("white musk", inplace=True)
    Perfumes_df.Concentration.replace("white musk.", "NaN", inplace=True)

In [393]:
if (Perfumes_df.Size == "-").any():
    Perfumes_df.Size.replace("-", "35 ml", inplace=True)
    Perfumes_df.Year.replace("Vanilla, Honey, Lemon, Jasmine, Mandarin, Magnolia, Passionfruit, Bergamot, Amber, Sandalwood, Patchouli", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Citron, jasmine, teak", "Vanilla, Honey, Lemon, Jasmine, Mandarin, Magnolia, Passionfruit, Bergamot, Amber, Sandalwood, Patchouli", inplace=True)
    Perfumes_df.Top_note.replace("-", "Vanilla, Honey, Lemon, Jasmine", inplace=True)
    Perfumes_df.Middle_note.replace("35 ml", "Mandarin, Magnolia, Passionfruit, Bergamot", inplace=True)
    Perfumes_df.Base_note.fillna("Amber, Sandalwood, Patchouli", inplace=True)

In [394]:
if (Perfumes_df.Size == "Modern").any():
    Perfumes_df.Size.replace("Modern", "75 ml", inplace=True)
    Perfumes_df.Year.replace("Floral", "2013", inplace=True)
    Perfumes_df.Ingredients.replace("2013", "Black violets, saffron, honey, lavender, cedar", inplace=True)
    Perfumes_df.Gender.replace("Eau de Toilette", "Men", inplace=True)
    Perfumes_df.Brand.replace("75 ml", "Roberto Cavalli", inplace=True)
    Perfumes_df.Character.replace("Men", "Modern", inplace=True)
    Perfumes_df.Concentration.replace("Black violets, saffron, honey, lavender, cedar", "Eau de Toilette", inplace=True)
    Perfumes_df.Product_Type.replace("Roberto Cavalli", "Perfume Set", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Perfume Set", "Floral", inplace=True)

In [395]:
if (Perfumes_df.Size == "Ginger, Bergamot, Orange").any():
    Perfumes_df.Size.replace("Ginger, Bergamot, Orange", "100 ml", inplace=True)
    Perfumes_df.Year.replace("Gardenia, Orange blossom", "NaN", inplace=True)
    Perfumes_df.Gender.replace("Eau de Parfum", "Women", inplace=True)
    Perfumes_df.Brand.replace("100 ml", "Valentino", inplace=True)
    Perfumes_df.Character.replace("Women", "NaN", inplace=True)
    Perfumes_df.Product_Type.replace("Valentino", "NaN", inplace=True)
    Perfumes_df.Ingredients.fillna("Oakmoss, Sandalwood, Musk, Tonka Bean, Vanilla", inplace=True)
    Perfumes_df.Top_note.fillna("Ginger, Bergamot, Orange", inplace=True)
    Perfumes_df.Middle_note.fillna("Gardenia, Orange blossom", inplace=True)
    Perfumes_df.Base_note.fillna("Oakmoss, Sandalwood, Musk, Tonka Bean, Vanilla", inplace=True)


In [396]:
if (Perfumes_df.Size == "Peony Flower").any():
    Perfumes_df.Brand.replace("200 ml", "Al Majed For Oud", inplace=True)
    Perfumes_df.Gender.replace("Al Majed For Oud", "NaN", inplace=True)
    Perfumes_df.Size.replace("Peony Flower", "200 ml", inplace=True)
    Perfumes_df.Year.replace("Caramel", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("vanilla", "Caramel, Vanilla", inplace=True)
    Perfumes_df.Top_note.replace("Ginger, Bergamot, Orange", "NaN", inplace=True)

In [397]:
if (Perfumes_df.Product_Type == "Room Spray").any():
    Perfumes_df.Size.fillna("200 ml", inplace=True)

In [398]:
if (Perfumes_df.Size == "Flowers, Jasmine").any():
    Perfumes_df.Size.replace("Flowers, Jasmine", "200 ml", inplace=True)
    Perfumes_df.Year.replace("Citrus", "NaN" , inplace=True)
    Perfumes_df.Ingredients.replace("Musk", "Flowers, Jasmine, Citrus, Musk", inplace=True)


In [399]:
if (Perfumes_df.Size == "Black tea, clary sage, guaiac wood").any():
    Perfumes_df.Brand.replace("300 ml","Zenology", inplace=True)
    Perfumes_df.Gender.replace("Zenology","NaN", inplace=True)
    Perfumes_df.Size.replace("Black tea, clary sage, guaiac wood","300 ml", inplace=True)
    Perfumes_df.Ingredients.replace("Oakmoss, Sandalwood, Musk, Tonka Bean, Vanilla","Black tea, clary sage, guaiac wood", inplace=True)


In [400]:
if (Perfumes_df.Size == "Alcohol, Propylene Glycol, Aqua (Water), Palmitic Acid, Stearic Acid, Ethylhexylglycerin, Parfum (Fragrance), Sodium Hydroxide, Hamamelis Virginiana (Witch Hazel) Water, Menthyl Lactate, Phenoxyethurate Extract, Salvia Office (SAGE) Leaf Extract, BHT, toc").any():
    Perfumes_df.Brand.replace("Eau de Parfum","Loewe", inplace=True)
    Perfumes_df.Gender.replace("Loewe", "Men", inplace=True)
    Perfumes_df.Product_Type.replace("Men", "Perfume Set", inplace=True)
    Perfumes_df.Size.replace("Alcohol, Propylene Glycol, Aqua (Water), Palmitic Acid, Stearic Acid, Ethylhexylglycerin, Parfum (Fragrance), Sodium Hydroxide, Hamamelis Virginiana (Witch Hazel) Water, Menthyl Lactate, Phenoxyethurate Extract, Salvia Office (SAGE) Leaf Extract, BHT, toc", "100 ml", inplace=True)
    Perfumes_df.Year.replace("Bergamot, Lemon, Tarragon", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Lavender, Geranium, Nutmeg", "Alcohol, Propylene Glycol, Aqua (Water), Palmitic Acid, Stearic Acid, Ethylhexylglycerin, Parfum (Fragrance), Sodium Hydroxide, Hamamelis Virginiana (Witch Hazel) Water, Menthyl Lactate, Phenoxyethurate Extract, Salvia Office (SAGE) Leaf Extract, BHT, toc", inplace=True)
    Perfumes_df.Concentration.replace("Sandalwood, Amber, Musk, Vetiver", "NaN", inplace=True)
    Perfumes_df.Top_note.fillna("Bergamot, Lemon, Tarragon", inplace=True)
    Perfumes_df.Middle_note.fillna("Lavender, Geranium, Nutmeg", inplace=True)
    Perfumes_df.Base_note.fillna("Sandalwood, Amber, Musk, Vetiver", inplace=True)

In [401]:
if (Perfumes_df.Size == "Ozone Notes, Passion Fruit, Agarwood, Amber").any():
    Perfumes_df.Fragrance_Family.replace("75 ml", "NaN", inplace=True)
    Perfumes_df.Size.replace("Ozone Notes, Passion Fruit, Agarwood, Amber", "75 ml", inplace=True)
    Perfumes_df.Year.replace("Eau de Parfum", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Black tea, clary sage, guaiac wood", "Ozone Notes, Passion Fruit, Agarwood, Amber", inplace=True)

In [402]:
Perfumes_df.Size.value_counts()

100 ml                                            2159
200 ml                                            1301
50 ml                                              883
75 ml                                              321
55 ml                                              149
                                                  ... 
14 ml                                                1
Orange, Saffron, Caramel                             1
Bergamot, Pear, Pink peppers, Maryam's incense       1
Grapefruit ,Guava                                    1
Amber, geranium, precious spices, dark woods         1
Name: Size, Length: 229, dtype: int64

In [403]:
if (Perfumes_df.Size == "Orange, Saffron, Caramel").any():
    Perfumes_df.Size.replace("Orange, Saffron, Caramel", "200 ml", inplace=True)
    Perfumes_df.Year.replace("Dry amber", "NaN", inplace=True)    
    Perfumes_df.Ingredients.replace("Vanilla, lavender", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Orange, Saffron, Caramel", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Dry amber", inplace=True)
    Perfumes_df.Base_note.replace("NaN","Vanilla, lavender", inplace=True)

In [404]:
if (Perfumes_df.Size == "Bergamot, Pear, Pink peppers, Maryam's incense").any():
    Perfumes_df.Size.replace("Bergamot, Pear, Pink peppers, Maryam's incense", "200 ml", inplace=True)
    Perfumes_df.Year.replace("Flower of Fawanya, Bulgarian Rose, Frisia Flower, Jasmine", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Perfume of honey, Amber, Patchouli", "Bergamot, Pear, Pink peppers, Maryam's incense,Flower of Fawanya, Bulgarian Rose, Frisia Flower, Jasmine, honey, Amber, Patchouli", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Bergamot, Pear, Pink peppers, Maryam's incense", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Flower of Fawanya, Bulgarian Rose, Frisia Flower, Jasmine", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Perfume of honey, Amber, Patchouli", inplace=True)

In [405]:
if (Perfumes_df.Size == "Grapefruit ,Guava").any():
    Perfumes_df.Size.replace("Grapefruit ,Guava", "200 ml", inplace=True)
    Perfumes_df.Ingredients.replace("Pineapple, Rose, Fawanya", "Pineapple, Rose, Fawanya,Dry amber,Pineapple, Rose, Fawanya", inplace=True)
    Perfumes_df.Top_note.replace("Orange, Saffron, Caramel", "Grapefruit ,Guava", inplace=True)
    Perfumes_df.Base_note.replace("Vanilla, lavender", "Pineapple, Rose, Fawanya", inplace=True)



In [406]:
if (Perfumes_df.Size == "Amber, geranium, precious spices, dark woods").any():
    Perfumes_df.Size.replace("Amber, geranium, precious spices, dark woods", "300 ml", inplace=True)
    Perfumes_df.Ingredients.replace("Ozone Notes, Passion Fruit, Agarwood, Amber", "Amber, geranium, precious spices, dark woods", inplace=True)
    Perfumes_df.Top_note.replace("Grapefruit ,Guava", "NaN", inplace=True)
    Perfumes_df.Middle_note.replace("Dry amber", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Pineapple, Rose, Fawanya", "NaN", inplace=True)


In [407]:
if (Perfumes_df.Size == "Apple, Cinnamon, Saffron, Sandalwood, Agarwood").any():
    Perfumes_df.Fragrance_Family.replace("12 ml", "NaN", inplace=True)
    Perfumes_df.Size.replace("Apple, Cinnamon, Saffron, Sandalwood, Agarwood", "12 ml",inplace=True )
    Perfumes_df.Ingredients.replace("Amber, geranium, precious spices, dark woods", "Apple, Cinnamon, Saffron, Sandalwood, Agarwood", inplace=True)


In [408]:
Perfumes_df.Size.value_counts()

100 ml                                                             2159
200 ml                                                             1304
50 ml                                                               883
75 ml                                                               321
55 ml                                                               149
                                                                   ... 
Sage and Saffron                                                      1
Cambodian Oud Oil                                                     1
Pepper, Pink Pepper, Geranium Leaves, Patchouli, Raspberry Rose       1
Pink Pepper, Violet, Mint Leaves                                      1
Extrait de Parfum                                                     1
Name: Size, Length: 224, dtype: int64

In [409]:
if (Perfumes_df.Size == "Pepper, Pink Pepper, Geranium Leaves, Patchouli, Raspberry Rose").any():
    Perfumes_df.Size.replace("Pepper, Pink Pepper, Geranium Leaves, Patchouli, Raspberry Rose", "75 ml", inplace=True)
    Perfumes_df.Brand.replace("Loewe", "Hermes", inplace=True)
    Perfumes_df.Gender.replace("Hermes", "NaN", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Grapefruit, Orange", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Apple, Cinnamon, Saffron, Sandalwood, Agarwood", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Grapefruit, Orange", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Pepper, Pink Pepper, Geranium Leaves, Patchouli, Raspberry Rose", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Atlantic Cedar, Vetiver, Resin", inplace=True)

In [410]:
if (Perfumes_df.Size == "Extrait de Parfum").any():
    Perfumes_df.Size.replace("Extrait de Parfum", "100 ml", inplace=True)

In [411]:
if (Perfumes_df.Size == "Sage and Saffron").any():
    Perfumes_df.Size.replace("Sage and Saffron", "50 g", inplace=True)
    Perfumes_df.Product_Type.replace("Charismatic", "NaN", inplace=True)
    Perfumes_df.Character.replace("Oriental", "Charismatic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Damask rose", "Oriental",inplace=True)
    Perfumes_df.Year.replace("Sandalwood and Musk", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("NaN", "Damask rose, Sage and Saffron, Sandalwood and Musk", inplace=True)
    Perfumes_df.Top_note.replace("Grapefruit, Orange", "Damask rose", inplace=True)
    Perfumes_df.Ingredients.replace("Pepper, Pink Pepper, Geranium Leaves, Patchouli, Raspberry Rose", "Sage and Saffron", inplace=True)
    Perfumes_df.Ingredients.replace("Atlantic Cedar, Vetiver, Resin", "Sandalwood and Musk", inplace=True)

In [412]:
if (Perfumes_df.Size == "Orange, Grapefruit, Benzoin, Wood notes, oak moss, Volcanic stones").any():
    Perfumes_df.Size.replace("Orange, Grapefruit, Benzoin, Wood notes, oak moss, Volcanic stones", "50 ml", inplace=True)
    Perfumes_df.Fragrance_Family.replace("50 ml","NaN" ,inplace=True)
    Perfumes_df.Ingredients.replace("Orange, Grapefruit", "Orange, Grapefruit, Benzoin, Wood notes, oak moss, Volcanic stones", inplace=True)
    Perfumes_df.Concentration.replace("Volcanic stones", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Benzoin, Wood notes, Oak moss", "Orange, Grapefruit", inplace=True)
    Perfumes_df.Middle_note.replace("Pepper, Pink Pepper, Geranium Leaves, Patchouli, Raspberry Rose", "Volcanic stones", inplace=True)
    Perfumes_df.Base_note.replace("Atlantic Cedar, Vetiver, Resin", "Benzoin, Wood notes, Oak moss", inplace=True)

In [413]:
if (Perfumes_df.Size == "Cambodian Oud Oil").any():
    Perfumes_df.Size.replace("Cambodian Oud Oil", "NaN", inplace=True)
    Perfumes_df.Year.replace("Perfume Oil", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Damask rose, Sage and Saffron, Sandalwood and Musk", "Cambodian Oud Oil", inplace=True)
    Perfumes_df.Top_note.replace("Damask rose", "NaN", inplace=True)
    Perfumes_df.Middle_note.replace("Volcanic stones", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Benzoin, Wood notes, Oak moss", "NaN", inplace=True)

In [414]:
if (Perfumes_df.Size == "1997").any():
    Perfumes_df.Size.replace("1997", "195 ml", inplace=True)
    Perfumes_df.Year.replace("Apple, Orange Blossom, Sweet Bergamot, Sparkling Lemon", "1997", inplace=True)
    Perfumes_df.Ingredients.replace("Rose, Patchouli, Teakwood", "Rose, Patchouli, Teakwood,Vanilla, Musk, Labdanum,Apple, Orange Blossom, Sweet Bergamot, Sparkling Lemon", inplace=True)
    Perfumes_df.Concentration.replace("Vanilla, Musk, Labdanum", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Apple, Orange Blossom, Sweet Bergamot, Sparkling Lemon", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Rose, Patchouli, Teakwood", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Vanilla, Musk, Labdanum", inplace=True)


In [415]:
if (Perfumes_df.Size == "Pink Pepper, Violet, Mint Leaves").any():
    Perfumes_df.Size.replace("Pink Pepper, Violet, Mint Leaves", "90 ml", inplace=True)
    Perfumes_df.Character.replace("90 ml","NaN", inplace=True)    
    Perfumes_df.Fragrance_Family.replace("Eau de Toilette", "NaN", inplace=True)    
    Perfumes_df.Year.replace("Sage, Watermelon, Pineapple, Lavender, Cinnamon", "NaN", inplace=True)    
    Perfumes_df.Concentration.replace("NaN", "Eau de Toilette", inplace=True)
    Perfumes_df.Top_note.replace("Apple, Orange Blossom, Sweet Bergamot", "Pink Pepper, Violet, Mint Leaves", inplace=True)
    Perfumes_df.Middle_note.replace("Rose, Patchouli, Teakwood", "Sage, Watermelon, Pineapple, Lavender, Cinnamon", inplace=True)
    Perfumes_df.Base_note.replace("Vanilla, Musk, Labdanum", "Vanilla, Chestnut, Cedar, Amber, Agarwood, Guaiac Wood", inplace=True)
    Perfumes_df.Ingredients.replace("Vanilla, Chestnut, Cedar, Amber, Agarwood, Guaiac Wood", "Vanilla, Chestnut, Cedar, Amber, Agarwood, Guaiac Wood,Pink Pepper, Violet, Mint Leaves,Sage, Watermelon, Pineapple, Lavender, Cinnamon", inplace=True)

In [416]:
Perfumes_df.Size.value_counts()

100 ml            2160
200 ml            1304
50 ml              884
75 ml              322
55 ml              149
                  ... 
14 ml                1
50 g                 1
195 ml               1
Floral,Chypre        1
Eau de Cologne       1
Name: Size, Length: 219, dtype: int64

In [417]:
if (Perfumes_df.Size == "Floral,Chypre").any():
    Perfumes_df.Size.replace("Floral,Chypre", "80 ml", inplace=True)
    Perfumes_df.Brand.replace("80 ml", "One Word",inplace=True)
    Perfumes_df.Gender.replace("One Word", "Unisex", inplace=True)
    Perfumes_df.Product_Type.replace("Unisex", "Hair & Body Mist", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Charismatic,Modern", "Floral,Chypre", inplace=True)
    Perfumes_df.Character.replace("Hair & Body Mist", "Charismatic,Modern", inplace=True)
    Perfumes_df.Concentration.replace("Bergamot", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("Orange Flower", "Bergamot", inplace=True)
    Perfumes_df.Middle_note.replace("Myrrh", "Orange Flower", inplace=True)
    Perfumes_df.Base_note.replace("Vanilla, Chestnut, Cedar, Amber, Agarwood, Guaiac Wood", "Myrrh", inplace=True)

In [418]:
if (Perfumes_df.Size == "Freesia, Amber, Musk").any():
    Perfumes_df.Size.replace("Freesia, Amber, Musk", "100 ml", inplace=True)
    Perfumes_df.Ingredients.replace("Freesia", "Freesia, Amber, Musk", inplace=True)
    Perfumes_df.Top_note.replace("Musk", "Freesia", inplace=True)
    Perfumes_df.Middle_note.replace("Sage, Watermelon, Pineapple, Lavender, Cinnamon", "Amber", inplace=True)
    Perfumes_df.Base_note.replace("Myrrh", "Musk", inplace=True)
    Perfumes_df.Concentration.replace("Amber", "NaN", inplace=True)


In [419]:
if (Perfumes_df.Size == "Eau de Cologne").any():
    Perfumes_df.Size.replace("Eau de Cologne", "50 ml", inplace=True)
    Perfumes_df.Character.replace("50 ml","NaN" , inplace=True)
    Perfumes_df.Concentration.replace("Leather accord, Cedar Wood, Patchouli, Benzoin", "Eau de Cologne",  inplace=True)
    Perfumes_df.Ingredients.replace("Leather accord, Cedar Wood, Patchouli, Benzoin, Musk", "Cardamom, Calabrian Bergamot, Sicilian Lemon, Ginger, Myrtle, Artemisia, Neroli, Leather accord, Cedar Wood, Patchouli, Benzoin, Musk", inplace=True)
    Perfumes_df.Top_note.replace("Apple, Orange Blossom, Sweet Bergamot", "Cardamom, Calabrian Bergamot, Sicilian Lemon, Ginger", inplace=True)
    Perfumes_df.Base_note.replace("Musk", "Leather accord, Cedar Wood, Patchouli, Benzoin, Musk", inplace=True)
    Perfumes_df.Middle_note.replace("Amber", "Myrtle, Artemisia, Neroli", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Cardamom, Calabrian Bergamot, Sicilian Lemon", "NaN", inplace=True)

In [420]:
if (Perfumes_df.Size == "Green Lemon, Ginger, White Pepper").any():
    Perfumes_df.Size.replace("Green Lemon, Ginger, White Pepper", "25 ml", inplace=True)
    Perfumes_df.Brand.replace("25 ml","Dr. Vranjes Firenze",  inplace=True)
    Perfumes_df.Gender.replace("Dr. Vranjes Firenze", "NaN", inplace=True)
    Perfumes_df.Year.replace("Green Lemon", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Ginger", "Green Lemon, Ginger, White Pepper", inplace=True)
    Perfumes_df.Concentration.replace("White Pepper","NaN",  inplace=True)
    Perfumes_df.Top_note.replace("Apple, Orange Blossom, Sweet Bergamot", "Green Lemon", inplace=True)
    Perfumes_df.Middle_note.replace("Myrtle, Artemisia, Neroli", "Ginger", inplace=True)
    Perfumes_df.Base_note.replace("eather accord, Cedar Wood, Patchouli", "White Pepper", inplace=True)


In [421]:
if (Perfumes_df.Size == "citrus fruits").any():
    Perfumes_df.Size.replace("citrus fruits", "50 g", inplace=True)
    Perfumes_df.Product_Type.replace("Extravagant", "Bakhoor", inplace=True)    
    Perfumes_df.Character.replace("Citrus", "Extravagant", inplace=True)    
    Perfumes_df.Fragrance_Family.replace("Citrus, rose, fruits, amber, sandalwood, musk", "Citrus", inplace=True)    
    Perfumes_df.Year.replace("Roses, fruits", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Amber, sandalwood, musk", "Citrus, rose, fruits, amber, sandalwood, musk", inplace=True)
    Perfumes_df.Top_note.replace("Apple, Orange Blossom, Sweet Bergamot", "citrus fruits", inplace=True)
    Perfumes_df.Middle_note.replace("Ginger", "Roses, fruits", inplace=True)
    Perfumes_df.Base_note.replace("Leather accord, Cedar Wood, Patchouli", "Amber, sandalwood, musk", inplace=True)

In [422]:
if (Perfumes_df.Size == "Pink Pepper, Jasmine, Amber, Patchouli").any():
    Perfumes_df.Size.replace("Pink Pepper, Jasmine, Amber, Patchouli", "35 ml", inplace=True)
    Perfumes_df.Year.replace("Pink Pepper, Jasmine,", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Amber", "Pink Pepper, Jasmine, Amber, Patchouli", inplace=True)
    Perfumes_df.Top_note.replace("grapefruit", "Pink Pepper, Jasmine,", inplace=True)
    Perfumes_df.Middle_note.replace("Roses, fruits", "Amber", inplace=True)
    Perfumes_df.Base_note.replace("Leather accord, Cedar Wood, Patchouli", "Patchouli", inplace=True)

- Cleaning Gender

In [423]:
if (Perfumes_df.Gender == "Bakhoor").any():
    Perfumes_df.Gender.replace("Bakhoor", "Home", inplace=True)

In [424]:
if (Perfumes_df.Gender == "Room Spray").any():
    Perfumes_df.Gender.replace("Room Spray", "Home", inplace=True)

In [425]:
if (Perfumes_df.Gender == "Candle").any():
    Perfumes_df['Gender'].replace("Candle", "Home", inplace=True)

In [426]:
if (Perfumes_df.Gender == "Victoria's Secret").any():
    Perfumes_df.Gender.replace("Victoria's Secret", "Women", inplace=True)

In [428]:
if (Perfumes_df.Gender == "Diffuser").any():
    Perfumes_df.Gender.replace("Diffuser", "Home", inplace=True)

- This is a shaving cream, thus I drop it 

In [697]:
Perfumes_df.drop(Perfumes_df.index[Perfumes_df['Gender'] == "Creed"], inplace=True)

In [698]:
if (Perfumes_df.Gender == "Perfume Set").any():
    Perfumes_df.Gender.replace("Perfume Set", "Women", inplace=True)

In [699]:
if (Perfumes_df.Gender == "Alrokon Alomani").any():
    Perfumes_df.Gender.replace("Alrokon Alomani", "Home", inplace=True)

In [ ]:
if (Perfumes_df.Name == "COACH SET FOR WOMEN (EDP 100 ML + EDP 7.5 ML + SHOWER GEL 100 ML)").any():
    Perfumes_df.Gender.replace("Coach", "Women", inplace=True)
    Perfumes_df.Character.replace("Romantic", "Feminine", inplace=True)
    Perfumes_df.Size.replace("Rasepberry Leaf, Pink Pepper, Pear,Turkish Rose,Musk, Sandalwood", "100 ml", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Feminine", "Fruity", inplace=True)
    Perfumes_df.Ingredients.replace("Tuberose, Iris, Jasmine, Aromatic Ingredients", "Rasepberry Leaf, Pink Pepper, Pear,Turkish Rose,Musk, Sandalwood", inplace=True)
    Perfumes_df.Concentration.replace("NaN", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Incense", "Rasepberry Leaf, Pink Pepper, Pear", inplace=True)
    Perfumes_df.Middle_note.replace("Flowers", "Turkish Rose", inplace=True)
    Perfumes_df.Base_note.replace("Jasmine, Aromatic Ingredients", "Musk, Sandalwood", inplace=True)

In [737]:
if (Perfumes_df.Name == "COACH SET FOR WOMEN (EDP 100 ML + EDP 7.5 ML + SHOWER GEL 100 ML)").any():
    Perfumes_df.Gender.replace("Coach", "Women", inplace=True)
    Perfumes_df.Character.replace("Romantic", "Feminine", inplace=True)
    Perfumes_df.Size.replace("Rasepberry Leaf, Pink Pepper, Pear,Turkish Rose,Musk, Sandalwood", "100 ml", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Feminine", "Fruity", inplace=True)
    Perfumes_df.Ingredients.replace("Tuberose, Iris, Jasmine, Aromatic Ingredients", "Rasepberry Leaf, Pink Pepper, Pear,Turkish Rose,Musk, Sandalwood", inplace=True)
    Perfumes_df.Concentration.replace("NaN", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Incense", "Rasepberry Leaf, Pink Pepper, Pear", inplace=True)
    Perfumes_df.Middle_note.replace("Flowers", "Turkish Rose", inplace=True)
    Perfumes_df.Base_note.replace("Jasmine, Aromatic Ingredients", "Musk, Sandalwood", inplace=True)

In [692]:
if (Perfumes_df.Name == "Deem Perfumes Hotel - 300 ml").any():
    Perfumes_df.Gender.replace('Deem Perfumes', "Home", inplace=True)
    Perfumes_df.Character.replace('Incense, Flowers, Amber, Woods', "NaN", inplace=True)
    Perfumes_df.Size.replace('100 ml', "300 ml", inplace=True)
    Perfumes_df.Ingredients.replace('Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine, Patchouli, Musk', "Incense, Flowers, Amber, Woods", inplace=True)
    Perfumes_df.Top_note.replace('Tuberose', "Incense", inplace=True)
    Perfumes_df.Middle_note.replace('Iris', "Flowers", inplace=True)
    Perfumes_df.Top_note.replace('Sandalwood,Cedarwood, Vanilla', "Amber, Woods", inplace=True)

In [691]:
if (Perfumes_df.Name == "Deem Perfumes Garden - 300 ml").any():
    #Perfumes_df.Gender.replace('Deem Perfumes', "Home", inplace=True)
    #Perfumes_df.Character.replace('Tuberose, Iris, Jasmine, Aromatic Ingredients', "NaN", inplace=True)
    #Perfumes_df.Size.replace('100 ml', "300 ml", inplace=True)
    Perfumes_df.Ingredients.replace('Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine, Patchouli, Musk', "Tuberose, Iris, Jasmine, Aromatic Ingredients", inplace=True)
    #Perfumes_df.Top_note.replace('Chamomile, Bitter Almond,Musk', "Tuberose", inplace=True)
    #Perfumes_df.Middle_note.replace('Indian Jasmine, Jasmine', "Iris", inplace=True)
    #Perfumes_df.Base_note.replace('Sandalwood,Cedarwood, Vanilla', "Jasmine, Aromatic Ingredients", inplace=True)

In [683]:
if (Perfumes_df.Name == "Gucci Bloom Set (EDP 100 ml + EDP 7.4 Rollerball + Perfumes Soap 100 g )").any():
    Perfumes_df.Gender.replace("Unisex", "Women", inplace=True)
    Perfumes_df.Character.replace("Charismatic", "Romantic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Romantic", "Floral", inplace=True)
    Perfumes_df.Concentration.replace("Jasmine absolute, Jasmine bud extract, Rangoon creeper", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Tuberose", "Jasmine absolute, Jasmine bud extract, Rangoon creeper", inplace=True)
    Perfumes_df.Middle_note.replace("Rangoon creeper", "Tuberose", inplace=True)
    Perfumes_df.Middle_note.replace("Sandalwood,Cedarwood, Vanilla", "Rangoon creeper", inplace=True)


In [680]:
if (Perfumes_df.Name == "Gucci Memoire D'Une Odeur Set (EDP 100 ml + EDP 7.4 ml)").any():
    Perfumes_df.Gender.replace("Gucci", "Unisex", inplace=True)
    Perfumes_df.Character.replace("Natural", "Charismatic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Oriental", "Woody", inplace=True)
    Perfumes_df.Year.replace("Chamomile, Bitter Almond", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Musk, Indian Jasmine, Jasmine", "Chamomile, Bitter Almond,Musk, Indian Jasmine, Jasmine, Sandalwood,Cedarwood, Vanilla", inplace=True)
    Perfumes_df.Concentration.replace("Sandalwood, Cedarwood, Vanilla","Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Chamomile, Bitter Almond,Musk", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Indian Jasmine, Jasmine", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Sandalwood,Cedarwood, Vanilla", inplace=True)


In [677]:
if (Perfumes_df.Gender == "Paco Rabanne").any():
    Perfumes_df.Gender.replace("Paco Rabanne", "Men", inplace=True)
    Perfumes_df.Size.replace("7 ml", "100 ml", inplace=True)
    Perfumes_df.Concentration.replace("NaN", "Eau de Toilette", inplace=True)
    Perfumes_df.Base_note.replace("Black Amber, Patchouli, Vanilla", "NaN", inplace=True)
    

In [675]:
if (Perfumes_df.Gender == "Byredo").any():
    Perfumes_df.Gender.replace("Byredo", "Unisex", inplace=True)
    Perfumes_df.Character.replace("Hair Mist", "Charismatic", inplace=True)
    Perfumes_df.Concentration.replace("Juniper Berries, Black Pepper", "Perfume", inplace=True)
    Perfumes_df.Top_note.replace("Violet, Iris, Leather", "Juniper Berries, Black Pepper", inplace=True)
    Perfumes_df.Middle_note.replace("Black Amber, Patchouli, Vanilla", "Violet, Iris, Leather", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Black Amber, Patchouli, Vanilla", inplace=True)


In [672]:
if (Perfumes_df.Name == "Mercedes Benz Set EDT 4x7 ml").any():
    Perfumes_df.Character.replace("Sensual", "Natural", inplace=True)
    Perfumes_df.Size.replace("300 ml", "7 ml", inplace=True)
    Perfumes_df.Concentration.replace("NaN", "Eau de Toilette", inplace=True)
    Perfumes_df.Ingredients.replace("Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("Tea, Bergamot, Osmanthus", "NaN", inplace=True)
    Perfumes_df.Middle_note.replace("Orchids, Jasmine, Rose, Freesia", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("African Orange Flower, Patchouli, Musk, Vanilla", "NaN", inplace=True)

In [664]:
if (Perfumes_df.Name == "Viktor & Rolf Ladies Perfume Set - 4×7 ml").any():
    Perfumes_df.Gender.replace("Viktor & Rolf", "Women", inplace=True)
    Perfumes_df.Character.replace("NaN","Sensual", inplace=True)
    Perfumes_df.Year.replace("Tea, Bergamot, Osmanthus", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Orchids, Jasmine, Rose, Freesia", "Tea, Bergamot, Osmanthus, Orchids, Jasmine, Rose, Freesia, African Orange Flower, Patchouli, Musk, Vanilla", inplace=True)
    Perfumes_df.Concentration.replace("African Orange Flower, Patchouli, Musk, Vanilla", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Tea, Bergamot, Osmanthus", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Orchids, Jasmine, Rose, Freesia", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "African Orange Flower, Patchouli, Musk, Vanilla", inplace=True)

In [657]:
if (Perfumes_df.Gender == "Faz").any():
    Perfumes_df.Gender.replace("Faz", "Unisex", inplace=True)
    Perfumes_df.Fragrance_Family.replace("NaN", "Floral", inplace=True)
    Perfumes_df.Ingredients.replace("Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine", "NaN", inplace=True)
    Perfumes_df.Size.replace("100 g", "300 ml", inplace=True)

In [652]:
if (Perfumes_df.Gender == "Ajmal").any():
    Perfumes_df.Gender.replace("Ajmal", "Unisex", inplace=True)
    Perfumes_df.Brand.replace("100 g", "Ajmal", inplace=True)
    Perfumes_df.Character.replace("Natural", "NaN", inplace=True)
    Perfumes_df.Fragrance_Family.replace("NaN", "Oriental", inplace=True)
    Perfumes_df.Size.replace("100 ml", "100 g", inplace=True)
    Perfumes_df.Ingredients.replace("Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("Pear, Black Currant", "NaN", inplace=True)
    Perfumes_df.Middle_note.replace("Pepper, Frankincense, Geranium, Fern", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Musk, Tonka Bean, Moss, Vetiver, Amber, Cedar", "NaN", inplace=True)


In [649]:
if (Perfumes_df.Name == "Dunhill Icon Racing Red Set (EDP 100 ml + 30 ml + Shower Gel 90 ml)").any():
    Perfumes_df.Ingredients.replace("Grapefruit, Black Pepper, Fine Grain, Mandarin, Orange, Solar Notes", "Grapefruit, Black Pepper, Fine Grain, Mandarin, Orange, Solar Notes,Pepper, Frankincense, Geranium, Fern,Musk, Tonka Bean, Moss, Vetiver, Amber, Cedar", inplace=True)
    Perfumes_df.Concentration.replace("Pepper, Frankincense, Geranium, Fern", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Musk, Tonka Bean, Moss, Vetiver, Amber, Cedar", "Grapefruit, Black Pepper, Fine Grain, Mandarin, Orange, Solar Notes", inplace=True)
    Perfumes_df.Middle_note.replace("Coriander, Neroli, Cardamom, Sea Salts", "Pepper, Frankincense, Geranium, Fern", inplace=True)
    Perfumes_df.Base_note.replace("Patchouli, Amber, Sandalwood, Driftwood","Musk, Tonka Bean, Moss, Vetiver, Amber, Cedar",  inplace=True)

In [646]:
if (Perfumes_df.Name == "Dunhill Icon Racing Blue Set (EDP 100 ml + 30 ml + Shower Gel 90 ml)").any():
    Perfumes_df.Character.replace("Sensual", "Natural", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Natural", "Woody", inplace=True)
    Perfumes_df.Size.replace("Woody", "100 ml", inplace=True)
    Perfumes_df.Ingredients.replace("Ambroxan, Bergamot, Lime", "Ambroxan, Bergamot, Lime,Coriander, Neroli, Cardamom, Sea Salts,Patchouli, Amber, Sandalwood, Driftwood", inplace=True)
    Perfumes_df.Concentration.replace("Coriander, Neroli, Cardamom, Sea Salts", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Patchouli, Amber, Sandalwood, Driftwood", "Ambroxan, Bergamot, Lime", inplace=True)
    Perfumes_df.Middle_note.replace("Lily of the Valley, Strawberry, Red Berries, Licorice, Girls' Hair", "Coriander, Neroli, Cardamom, Sea Salts", inplace=True)
    Perfumes_df.Base_note.replace("Sandalwood, Tonka Bean, Musk, Vanilla, Caramel", "Patchouli, Amber, Sandalwood, Driftwood", inplace=True)

In [643]:
if (Perfumes_df.Gender == "Dunhill").any():
    Perfumes_df.Character.replace("Sensual", "Natural", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Natural", "Woody", inplace=True)
    Perfumes_df.Size.replace("Woody", "100 ml", inplace=True)
    Perfumes_df.Ingredients.replace("Ambroxan, Bergamot, Lime", "Ambroxan, Bergamot, Lime,Coriander, Neroli, Cardamom, Sea Salts,Patchouli, Amber, Sandalwood, Driftwood", inplace=True)
    Perfumes_df.Concentration.replace("Coriander, Neroli, Cardamom, Sea Salts", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Patchouli, Amber, Sandalwood, Driftwood", "Ambroxan, Bergamot, Lime", inplace=True)
    Perfumes_df.Middle_note.replace("Lily of the Valley, Strawberry, Red Berries, Licorice, Girls' Hair", "Coriander, Neroli, Cardamom, Sea Salts", inplace=True)
    Perfumes_df.Base_note.replace("Sandalwood, Tonka Bean, Musk, Vanilla, Caramel", "Patchouli, Amber, Sandalwood, Driftwood", inplace=True)


In [637]:
if (Perfumes_df.Gender == "Aquolina").any():
    Perfumes_df.Gender.replace("Aquolina", "Women", inplace=True)
    Perfumes_df.Character.replace("Charismatic", "Sensual", inplace=True)
    Perfumes_df.Concentration.replace("Lily of the Valley, Strawberry, Red Berries, Licorice, Girls' Hair", "Eau de Toilette", inplace=True)
    Perfumes_df.Ingredients.replace("Bergamot, Fig Leaf, Mandarin, Raspberry", "Bergamot, Fig Leaf, Mandarin, Raspberry, Lily of the Valley, Strawberry, Red Berries, Licorice, Girls' Hair, Sandalwood, Tonka Bean, Musk, Vanilla, Caramel", inplace=True)
    Perfumes_df.Top_note.replace("Sandalwood, Tonka Bean, Musk, Vanilla, Caramel", "Bergamot, Fig Leaf, Mandarin, Raspberry", inplace=True)
    Perfumes_df.Middle_note.replace("Almond, Tonka Bean, Patchouli", "Lily of the Valley, Strawberry, Red Berries, Licorice, Girls' Hair", inplace=True)
    Perfumes_df.Base_note.replace("Almond, Tonka Bean, Patchouli", "Sandalwood, Tonka Bean, Musk, Vanilla, Caramel", inplace=True)

In [635]:
if (Perfumes_df.Gender == "Tom Ford").any():
    Perfumes_df.Gender.replace("Tom Ford", "Women", inplace=True)
    Perfumes_df.Character.replace("Soft Floral", "Feminine", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Oriental", "Soft Floral", inplace=True)
    Perfumes_df.Size.replace("2011", "100 ml", inplace=True)
    Perfumes_df.Year.replace("2011","Iris Flowers, Jasmine & Iris Roots, Absolute Violet Leaves, Italian Mandarin and Rose Bay. Javanese, cedarwood, absolute vetiver, selkolid, fine suede.", "2011", inplace=True)
    Perfumes_df.Ingredients.replace("Eau de Parfum", "Iris Flowers, Jasmine & Iris Roots, Absolute Violet Leaves, Italian Mandarin and Rose Bay. Javanese, cedarwood, absolute vetiver, selkolid, fine suede.", inplace=True)
    Perfumes_df.Concentration.replace("Iris Flowers, Jasmine & and Iris Roots.","Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("Absolute Violet Leaves, Italian Mandarin and Rose Bay.", "Iris Flowers, Jasmine & and Iris Roots.", inplace=True)
    Perfumes_df.Middle_note.replace("Javanese, cedarwood, absolute vetiver, selkolid, fine suede.", "Absolute Violet Leaves, Italian Mandarin and Rose Bay.", inplace=True)
    Perfumes_df.Base_note.replace("Almond, Tonka Bean, Patchouli", "Javanese, cedarwood, absolute vetiver, selkolid, fine suede.", inplace=True)

In [631]:
if (Perfumes_df.Gender == "Alwaleef").any():
    Perfumes_df.Gender.replace("Alwaleef", "Unisex", inplace=True)
    Perfumes_df.Character.replace("Sensual", "Charismatic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Aromatic", "Oriental", inplace=True)
    Perfumes_df.Year.replace("Pear, Black Currant", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Orange Blossom, Jasmine, Iris", "Pear, Black Currant,Orange Blossom, Jasmine, Iris,Almond, Tonka Bean, Patchouli", inplace=True)
    Perfumes_df.Concentration.replace("Almond, Tonka Bean, Patchouli", "Eau de Parfum", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Pear, Black Currant", inplace=True)
    Perfumes_df.Middle_note.replace("Damask rose", "Almond, Tonka Bean, Patchouli", inplace=True)
    Perfumes_df.Base_note.replace("Sandalwood, Tolu Balsam, Almond, Vanilla,Tonka Bean", "Almond, Tonka Bean, Patchouli", inplace=True)

In [626]:
if (Perfumes_df.Gender == "Carrera Jeans").any():
    Perfumes_df.Gender.replace("Carrera Jeans", "Women", inplace=True)
    Perfumes_df.Character.replace("Charismatic", "Sensual", inplace=True)
    Perfumes_df.Concentration.replace("Damask rose", "Eau de Parfum", inplace=True)
    Perfumes_df.Ingredients.replace("Bitter Orange", "Bitter Orange,Damask rose, Sandalwood, Tolu Balsam, Almond, Vanilla,Tonka Bean", inplace=True)
    Perfumes_df.Top_note.replace("Sandalwood, Tolu Balsam, Almond, Vanilla,Tonka Bean", "Bitter Orange", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Damask rose", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Sandalwood, Tolu Balsam, Almond, Vanilla,Tonka Bean", inplace=True)

In [624]:
if (Perfumes_df.Gender == "Atyaab Al Noman").any():
    Perfumes_df.Gender.replace("Atyaab Al Noman", "Unisex", inplace=True)
    Perfumes_df.Brand.replace("40 g", "Atyaab Al Noman", inplace=True)
    Perfumes_df.Fragrance_Family.replace("NaN", "Oriental", inplace=True)
    Perfumes_df.Size.replace("200 ml", "40 g", inplace=True)
    Perfumes_df.Ingredients.replace("Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("Thyme, Bergamot, Apple, Basil", "NaN", inplace=True)
    Perfumes_df.Middle_note.replace("Hedione, Cardamom, Licorice", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Virginian Cedar, White Honey, Rose, Patchouli", "NaN", inplace=True)

In [622]:
if (Perfumes_df.Gender == "Caron").any():
    Perfumes_df.Gender.replace("Caron", "Men", inplace=True)
    Perfumes_df.Character.replace("Modern", "Charismatic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Oriental", "Aromatic", inplace=True)
    Perfumes_df.Size.replace("NaN", "200 ml", inplace=True)
    Perfumes_df.Concentration.replace("Lavender", "Eau de Toilette", inplace=True)
    Perfumes_df.Top_note.replace("Vanilla", "Lavender", inplace=True)
    Perfumes_df.Middle_note.replace("Amber, Musk", "Vanilla", inplace=True)
    Perfumes_df.Top_note.replace("Virginian Cedar, White Honey, Rose, Patchouli", "Amber, Musk", inplace=True)

In [616]:
if (Perfumes_df.Gender == "Elie Saab").any():
    Perfumes_df.Gender.replace("Elie Saab", "Women", inplace=True)
    Perfumes_df.Character.replace("Charismatic", "Modern", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Modern", "Floral Oriental", inplace=True)
    Perfumes_df.Size.replace("Floral Oriental", "90 ml", inplace=True)
    Perfumes_df.Concentration.replace("Orange Blossom", "Eau de Toilette", inplace=True)
    Perfumes_df.Top_note.replace("Jasmine", "Orange Blossom", inplace=True)
    Perfumes_df.Middle_note.replace("Virginian Cedar, White Honey, Rose, Patchouli", "Jasmine", inplace=True)
    Perfumes_df.Base_note.replace("Patchouli, Cedar, Amber Wood, Vetiver, Vanilla", "Virginian Cedar, White Honey, Rose, Patchouli", inplace=True)


In [614]:
if (Perfumes_df.Gender == "Diesel").any():
    Perfumes_df.Gender.replace("Diesel", "Men", inplace=True)
    Perfumes_df.Character.replace("Extravagant", "Charismatic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Charismatic", "Oriental", inplace=True)
    Perfumes_df.Size.replace("Oriental", "75 ml", inplace=True)
    Perfumes_df.Year.replace("Thyme, Bergamot, Apple, Basil", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Hedione, Cardamom, Licorice", "Thyme, Bergamot, Apple, Basil,Hedione, Cardamom, Licorice,Patchouli, Cedar, Amber Wood, Vetiver, Vanilla", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Thyme, Bergamot, Apple, Basil", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Hedione, Cardamom, Licorice", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Patchouli, Cedar, Amber Wood, Vetiver, Vanilla", inplace=True)


In [611]:
if (Perfumes_df.Gender == "Who Am I").any():
    Perfumes_df.Gender.replace("Who Am I", "Unisex", inplace=True)
    Perfumes_df.Character.replace("Bundles", "NaN", inplace=True)
    Perfumes_df.Size.replace("10 g", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine", "NaN", inplace=True)

In [609]:
if (Perfumes_df.Gender == "Cartier").any():
    Perfumes_df.Gender.replace("Cartier", "Unisex", inplace=True)
    Perfumes_df.Character.replace("Sensual", "Extravagant", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Extravagant", "Aromatic", inplace=True)
    Perfumes_df.Size.replace("Aromatic", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Vanilla, Honey, Lemon, Jasmine, Mandarin", "NaN", inplace=True)
    Perfumes_df.Concentration.replace("Eau de Toilette", "Eau de Perfum", inplace=True)
    Perfumes_df.Middle_note.replace("Rose", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Patchouli, Vetiver, Tonka bean, Moss", "NaN", inplace=True)



In [604]:
if (Perfumes_df.Gender == "Guerlain").any():
    Perfumes_df.Gender.replace("Guerlain" ,"Women", inplace=True)
    Perfumes_df.Brand.replace("Eau de Toilette", "Guerlain", inplace=True)
    Perfumes_df.Size.replace("Eau de Toilette" ,"50 ml", inplace=True)
    Perfumes_df.Year.replace("Bergamot, Raspberry, Rose, Patchouli, Vetiver, Tonka bean, Moss" ,"NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Bergamot, Raspberry" ,"Bergamot, Raspberry, Rose, Patchouli, Vetiver, Tonka bean, Moss", inplace=True)
    Perfumes_df.Concentration.replace("Rose" ,"Eau de Toilette", inplace=True)
    Perfumes_df.Top_note.replace("Patchouli, Vetiver, Tonka bean, Moss" ,"Bergamot, Raspberry", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Rose", inplace=True)
    Perfumes_df.Base_note.replace("NaN" ,"Patchouli, Vetiver, Tonka bean, Moss", inplace=True)

In [601]:
if (Perfumes_df.Gender == "Alsinan Perfumes").any():
    Perfumes_df.Gender.replace("Alsinan Perfumes", "Home", inplace=True)
    Perfumes_df.Brand.replace("10 g", "Alsinan Perfumes", inplace=True)
    Perfumes_df.Fragrance_Family.replace("NaN", "Oriental", inplace=True)
    Perfumes_df.Size.replace("250 ml", "10 g", inplace=True)
    Perfumes_df.Ingredients.replace("Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("Blackcurrant, Bergamot", "NaN", inplace=True)
    Perfumes_df.Middle_note.replace("Neroli, White Geranium, Jasmine", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Sandalwood, Patchouli, Oakmoss, Amberwwod, Musk", "NaN", inplace=True)

In [596]:
if (Perfumes_df.Gender == "Chloe").any():
    Perfumes_df.Gender.replace("Chloe", "Women", inplace=True)
    Perfumes_df.Year.replace("Lemon, Mirabelle, Bergamot, Orange, Freesia, Peach, Rose, Jasmine, Sandalwood, Patchouli, Oakmoss, Amberwwod, Musk", "NaN", inplace=True)    
    Perfumes_df.Ingredients.replace("Lemon, Mirabelle, Bergamot, Orange", "Lemon, Mirabelle, Bergamot, Orange, Freesia, Peach, Rose, Jasmine, Sandalwood, Patchouli, Oakmoss, Amberwwod, Musk", inplace=True)    
    Perfumes_df.Concentration.replace("Freesia, Peach, Rose, Jasmine", "Eau de Parfum", inplace=True)    
    Perfumes_df.Top_note.replace("Sandalwood, Patchouli, Oakmoss, Amberwwod, Musk", "Lemon, Mirabelle, Bergamot, Orange", inplace=True)    
    Perfumes_df.Middle_note.replace("Neroli, White Geranium, Jasmine", "Freesia, Peach, Rose, Jasmine", inplace=True)
    Perfumes_df.Base_note.replace("Patchouli, Musk", "Sandalwood, Patchouli, Oakmoss, Amberwwod, Musk", inplace=True)

In [583]:
Perfumes_df['Gender'].mask(Perfumes_df['Name'].str.contains("Spray"),"Home", inplace=True)

In [585]:
Perfumes_df['Gender'].mask(Perfumes_df['Name'].str.contains("Gift"),"Women", inplace=True)

In [587]:
Perfumes_df['Gender'].mask(Perfumes_df['Name'].str.contains("D'Hermes"),"Men", inplace=True)

In [589]:
Perfumes_df['Gender'].mask(Perfumes_df['Name'].str.contains("Hermes Un Jardin"),"Unisex", inplace=True)

In [433]:
if (Perfumes_df.Gender == "Perfume Oil").any():
    Perfumes_df.Brand.replace("3 ml", "Junaid", inplace=True)
    Perfumes_df.Gender.replace("Perfume Oil", "Unisex", inplace=True)
    Perfumes_df.Product_Type.replace("Junaid Perfumes", "Perfume Set,Perfume Oil", inplace=True)
    Perfumes_df.Character.replace("Unisex", "Dynamic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Perfume Set,Perfume Oil", "Arabian", inplace=True)
    Perfumes_df.Ingredients.replace("Cambodian Oud Oil", "NaN", inplace=True)
    Perfumes_df.Top_note.replace("Apple, Orange Blossom, Sweet Bergamot", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Amber", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Leather accord, Cedar Wood, Patchouli", "NaN", inplace=True)

In [700]:
Perfumes_df.drop(Perfumes_df.index[Perfumes_df['Gender'] == "Wadi Halfa"], inplace=True)

In [701]:
Perfumes_df.drop(Perfumes_df.index[Perfumes_df['Gender'] == "Car Air Freshener"], inplace=True)

In [702]:
Perfumes_df.drop(Perfumes_df.index[Perfumes_df['Gender'] == "Diffuser Refill"], inplace=True)

In [703]:
Perfumes_df.drop(Perfumes_df.index[2908], inplace=True)

In [748]:
Perfumes_df['Gender'].value_counts()

Women        4280
Men          1001
Home          633
Mabkhara       40
Bundles        38
             ... 
Coach           2
Xerjoff         2
Valentino       2
Lancome         2
Miu Miu         2
Name: Gender, Length: 62, dtype: int64

- Rate Column

In [435]:
Perfumes_df.Rate.value_counts()

none    6020
5        424
4         36
4.5       31
3         10
4.25       8
3.5        7
4.75       7
4.6        5
1          5
4.83       4
3.67       4
4.67       3
4.86       3
4.93       3
4.88       3
4.8        2
4.38       2
4.89       1
4.92       1
4.9        1
4.95       1
4.42       1
4.79       1
3.33       1
Name: Rate, dtype: int64

In [436]:
Perfumes_df.Rating_count.value_counts()

none         6020
1 Rating      330
2 Rating      111
3 Rating       52
4 Rating       26
5 Rating       11
6 Rating       10
8 Rating        6
7 Rating        4
18 Rating       3
13 Rating       2
17 Rating       1
12 Rating       1
Name: Rating_count, dtype: int64

As shown above I don't have a lot of rating data, thus Iam going to choose a content-based model that focuses on perfume features 

- Brand Column 

In [437]:
# How many Brand I have? 
print(f'This dataset consist of : {Perfumes_df.Brand.nunique()} brands')

This dataset consist of : 454 brands


- Product Type 

In [438]:
print(f'This dataset consist of : {Perfumes_df.Product_Type.nunique()} different product type')

This dataset consist of : 107 different product type


107 is a big number it not possible this column need cleaning

I found some product type that should be excluded like makeup, shower gel, body lotion 

In [749]:
Perfumes_df.Product_Type.value_counts()

Perfume               4564
Perfume Set            697
Room Spray             155
Perfume Oil            143
Candle                 136
Hair Mist              108
Bakhoor                106
Diffuser                93
Oud                     88
Body Mist               79
Hair & Body Mist        57
Clothes Fragrance       13
Bundles                  7
Car Air Freshener        4
Home Fragrance Set       1
Name: Product_Type, dtype: int64

In [752]:
Perfumes_df.to_csv("Cleaned_1.csv")

### Cleaning 

In [440]:
if (Perfumes_df.Name == "Room Spray").any():
    Perfumes_df.Product_Type.replace("Natural oud, Oriental Oils", "Room Spray",inplace=True)

In [441]:
if (Perfumes_df.Product_Type == "Oriental").any():
    Perfumes_df.Product_Type.replace("Oriental", "Perfume", inplace=True)

In [442]:
if (Perfumes_df.Product_Type == "Wild berries").any():
    Perfumes_df.Product_Type.replace("Wild berries", "Perfume", inplace=True)

In [443]:
if (Perfumes_df.Product_Type == "Mugler").any():
    Perfumes_df.Product_Type.replace("Mugler", "Perfume", inplace=True)

In [444]:
if (Perfumes_df.Product_Type == "Oud, French notes").any():
    Perfumes_df.Product_Type.replace("Oud, French notes", "Perfume", inplace=True)

In [445]:
if (Perfumes_df.Product_Type == "Chanel").any():
    Perfumes_df.Product_Type.replace("Chanel", "Perfume", inplace=True)

- Drop rows that is not relevant to my project

In [704]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Deodorant"].index
Perfumes_df.drop(index_names, inplace=True)

In [705]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Compact Powder"].index
Perfumes_df.drop(index_names, inplace=True)

In [706]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Face Cream"].index
Perfumes_df.drop(index_names, inplace=True)

In [707]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Shower Gel,Hair Gel"].index
Perfumes_df.drop(index_names, inplace=True)

In [708]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Shower Gel"].index
Perfumes_df.drop(index_names, inplace=True)

In [451]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Diffuser ,Home Fragrance"].index
Perfumes_df.drop(index_names, inplace=True)

In [709]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Body Cream"].index
Perfumes_df.drop(index_names, inplace=True)

In [720]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Body Lotion"].index
Perfumes_df.drop(index_names, inplace=True)


In [718]:
index_names = Perfumes_df[Perfumes_df.Product_Type == "Mabkhara"].index
Perfumes_df.drop(index_names, inplace=True)


In [721]:
Perfumes_df.shape

(6565, 18)

### Cleaning 

In [454]:
if (Perfumes_df.Product_Type == "Red berries, jasmine, vanilla, amber and musk").any():
    Perfumes_df.Product_Type.replace("Red berries, jasmine, vanilla, amber and musk", "Perfume", inplace=True)

In [455]:
if (Perfumes_df.Product_Type == "Armand Basi").any():
    Perfumes_df.Product_Type.replace("Armand Basi", "Perfume", inplace=True)

In [456]:
if (Perfumes_df.Product_Type == "Dynamic").any():
    Perfumes_df.Product_Type.replace("Dynamic", "Perfume", inplace=True)

In [457]:
if (Perfumes_df.Product_Type == "Cartier").any():
    Perfumes_df.Product_Type.replace("Cartier", "Perfume", inplace=True)

In [458]:
if (Perfumes_df.Product_Type == "Miu Miu").any():
    Perfumes_df.Product_Type.replace("Miu Miu", "Perfume", inplace=True)

In [459]:
if (Perfumes_df.Product_Type == "Musk").any():
    Perfumes_df.Product_Type.replace("Musk", "Perfume", inplace=True)

In [460]:
if (Perfumes_df.Product_Type == "Concealer").any():
    Perfumes_df.Product_Type.replace("Concealer", "Perfume", inplace=True)

In [461]:
if (Perfumes_df.Product_Type == "Mont Blanc").any():
    Perfumes_df.Product_Type.replace("Mont Blanc", "Perfume", inplace=True)

In [462]:
if (Perfumes_df.Product_Type == "Coach").any():
    Perfumes_df.Product_Type.replace("Coach", "Perfume", inplace=True)

In [463]:
if (Perfumes_df.Product_Type == "Dunhill").any():
    Perfumes_df.Product_Type.replace("Dunhill", "Perfume", inplace=True)

In [464]:
if (Perfumes_df.Product_Type == "Estée Lauder").any():
    Perfumes_df.Product_Type.replace("Estée Lauder", "Perfume", inplace=True)

In [465]:
if (Perfumes_df.Product_Type == "Hugo Boss").any():
    Perfumes_df.Product_Type.replace("Hugo Boss", "Perfume", inplace=True)

In [466]:
if (Perfumes_df.Product_Type == "Versace").any():
    Perfumes_df.Product_Type.replace("Versace", "Perfume", inplace=True)

In [467]:
if (Perfumes_df.Product_Type == "Creed").any():
    Perfumes_df.Product_Type.replace("Creed", "Perfume", inplace=True)

In [468]:
if (Perfumes_df.Product_Type == "Violet").any():
    Perfumes_df.Product_Type.replace("Violet", "Perfume", inplace=True)

In [469]:
if (Perfumes_df.Product_Type == "Lacoste").any():
    Perfumes_df.Product_Type.replace("Lacoste", "Perfume", inplace=True)

In [470]:
if (Perfumes_df.Product_Type == "Perfume Set,Perfume Oil").any():
    Perfumes_df.Product_Type.replace("Perfume Set,Perfume Oil", "Perfume Set", inplace=True)

In [471]:
if (Perfumes_df.Product_Type == "Lorenzo Villoresi Firenze").any():
    Perfumes_df.Product_Type.replace("Lorenzo Villoresi Firenze", "Perfume", inplace=True)

In [472]:
if (Perfumes_df.Product_Type == "Omani Frankincense, Mukhallat Malaki").any():
    Perfumes_df.Product_Type.replace("Omani Frankincense, Mukhallat Malaki", "Perfume", inplace=True)

In [473]:
if (Perfumes_df.Product_Type == "170 g").any():
    Perfumes_df.Product_Type.replace("170 g", "Oud", inplace=True)

In [474]:
if (Perfumes_df.Product_Type == "Natural").any():
    Perfumes_df.Product_Type.replace("Natural", "Perfume", inplace=True)

In [475]:
if (Perfumes_df.Product_Type == "Cotton").any():
    Perfumes_df.Product_Type.replace("Cotton", "Diffuser", inplace=True)

In [476]:
if (Perfumes_df.Product_Type == "Fruity scents").any():
    Perfumes_df.Product_Type.replace("Fruity scents", "Perfume", inplace=True)

In [477]:
if (Perfumes_df.Product_Type == "Omani Frankincense, lavender").any():
    Perfumes_df.Product_Type.replace("Omani Frankincense, lavender", "Perfume", inplace=True)

In [478]:
if (Perfumes_df.Product_Type == "Oriental notes, French notes").any():
    Perfumes_df.Product_Type.replace("Oriental notes, French notes", "Perfume", inplace=True)

In [479]:
if (Perfumes_df.Product_Type == "Cactus petals, Vanilla, Coconut, Lemon, Essential Oils").any():
    Perfumes_df.Product_Type.replace("Cactus petals, Vanilla, Coconut, Lemon, Essential Oils", "Candle", inplace=True)

In [480]:
if (Perfumes_df.Product_Type == "Cherry, Blackberry, Merlot, Perfume oils").any():
    Perfumes_df.Product_Type.replace("Cherry, Blackberry, Merlot, Perfume oils", "Perfume Oil", inplace=True)

In [481]:
if (Perfumes_df.Product_Type == "Coronet").any():
    Perfumes_df.Product_Type.replace("Coronet", "Room Spray", inplace=True)

In [482]:
if (Perfumes_df.Product_Type == "20 g").any():
    Perfumes_df.Product_Type.replace("20 g", "Bakhoor", inplace=True)

In [483]:
if (Perfumes_df.Product_Type == "Sparkling champagne, crystal peony, almond cream").any():
    Perfumes_df.Product_Type.replace("Sparkling champagne, crystal peony, almond cream", "Candle", inplace=True)
    Perfumes_df.Ingredients.replace("Chrysanthemum, pink vanilla, apple, cassis, woods", "Sparkling champagne, crystal peony, almond cream", inplace=True)

In [484]:
if (Perfumes_df.Product_Type == "Anfasic Dokhoon").any():
    Perfumes_df.Product_Type.replace("Anfasic Dokhoon", "Oud", inplace=True)
    Perfumes_df.Year.replace("Ozone Notes, Passion Fruit, Agarwood, Amber", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Chrysanthemum, pink vanilla, apple, cassis, woods", "Ozone Notes, Passion Fruit, Agarwood, Amber", inplace=True)

In [485]:
if (Perfumes_df.Product_Type == "Bed Linen Spray").any():
    Perfumes_df.Product_Type.replace("Bed Linen Spray", "Room Spray", inplace=True)

In [486]:
if (Perfumes_df.Product_Type == "Van Cleef & Arpels").any():
    Perfumes_df.Product_Type.replace("Van Cleef & Arpels", "Perfume", inplace=True)


In [487]:
if (Perfumes_df.Product_Type == "Chrysanthemum, pink vanilla, apple, cassis, woods").any():
    Perfumes_df.Product_Type.replace("Chrysanthemum, pink vanilla, apple, cassis, woods", "Candle", inplace=True)
    Perfumes_df.Ingredients.replace("Pineapple, papaya, mandarin orange, sugar cane and peach, musk, sandalwood and coconut", "Chrysanthemum, pink vanilla, apple, cassis, woods", inplace=True)

In [488]:
if (Perfumes_df.Product_Type == "Pineapple, papaya, mandarin orange, sugar cane and peach, musk, sandalwood and coconut.").any():
    Perfumes_df.Product_Type.replace("Pineapple, papaya, mandarin orange, sugar cane and peach, musk, sandalwood and coconut.", "Room Spray", inplace=True)
    Perfumes_df.Ingredients.replace("NaN", "Pineapple, papaya, mandarin orange, sugar cane and peach, musk, sandalwood and coconut", inplace=True)
    Perfumes_df.Top_note.replace("Apple, Orange Blossom, Sweet Bergamot, Sparkling Lemon", "NaN", inplace=True)
    Perfumes_df.Character.replace("500 ml", "NaN", inplace=True)
    Perfumes_df.Size.replace("200 ml", "500 ml", inplace=True)
    Perfumes_df.Middle_note.replace("Amber", "NaN", inplace=True)
    Perfumes_df.Base_note.replace("Leather accord, Cedar Wood, Patchouli, Benzoin, Musk", "NaN", inplace=True)


In [489]:
if (Perfumes_df.Product_Type == "Floral").any():
    Perfumes_df.Product_Type.replace("Floral", "Candle", inplace=True)

In [490]:
if (Perfumes_df.Product_Type == "Fruity").any():
    Perfumes_df.Product_Type.replace("Fruity", "Candle", inplace=True)

In [491]:
if (Perfumes_df.Product_Type == "Citrus").any():
    Perfumes_df.Product_Type.replace("Citrus", "Candle", inplace=True)

In [492]:
if (Perfumes_df.Product_Type == "Essential Oil").any():
    Perfumes_df.Product_Type.replace("Essential Oil", "Perfume Oil", inplace=True)

In [493]:
if (Perfumes_df.Product_Type == "500 ml").any():
    Perfumes_df.Product_Type.replace("500 ml", "Room Spray", inplace=True)

In [494]:
if (Perfumes_df.Product_Type == "Giorgio Armani").any():
    Perfumes_df.Product_Type.replace("Giorgio Armani", "Perfume", inplace=True)

In [495]:
if (Perfumes_df.Product_Type == "Gucci").any():
    Perfumes_df.Product_Type.replace("Gucci", "Perfume", inplace=True)

In [496]:
if (Perfumes_df.Product_Type == "Moschino").any():
    Perfumes_df.Product_Type.replace("Moschino", "Perfume Set", inplace=True)

In [497]:
if (Perfumes_df.Product_Type == "Burberry").any():
    Perfumes_df.Product_Type.replace("Burberry", "Perfume Set", inplace=True)

In [498]:
if (Perfumes_df.Product_Type == "Bvlgari").any():
    Perfumes_df.Product_Type.replace("Bvlgari", "Perfume Set", inplace=True)

In [499]:
if (Perfumes_df.Product_Type == "Woody Oriental").any():
    Perfumes_df.Product_Type.replace("Woody Oriental", "Bakhoor", inplace=True)

In [500]:
if (Perfumes_df.Product_Type == "Floral Oriental").any():
    Perfumes_df.Product_Type.replace("Floral Oriental", "Room Spray", inplace=True)

In [501]:
if (Perfumes_df.Product_Type == "Perfume,Body Mist").any():
    Perfumes_df.Product_Type.replace("Perfume,Body Mist", "Body Mist", inplace=True)

In [502]:
if (Perfumes_df.Product_Type == "Kids").any():
    Perfumes_df.Product_Type.replace("Kids", "Perfume Set", inplace=True)

In [503]:
if (Perfumes_df.Product_Type == "Aquatic").any():
    Perfumes_df.Product_Type.replace("Aquatic", "Candle", inplace=True)

In [504]:
if (Perfumes_df.Product_Type == "Dolce&Gabbana").any():
    Perfumes_df.Product_Type.replace("Dolce&Gabbana", "Perfume Set", inplace=True)

In [505]:
if (Perfumes_df.Product_Type == "Guerlain").any():
    Perfumes_df.Product_Type.replace("Guerlain", "Perfume Set", inplace=True)

In [506]:
if (Perfumes_df.Product_Type == "Calvin Klein").any():
    Perfumes_df.Product_Type.replace("Calvin Klein", "Perfume Set", inplace=True)

In [507]:
if (Perfumes_df.Product_Type == "Givenchy").any():
    Perfumes_df.Product_Type.replace("Givenchy", "Perfume Set", inplace=True)

In [508]:
if (Perfumes_df.Product_Type == "Diffuser ,Diffuser Refill").any():
    Perfumes_df.Product_Type.replace("Diffuser ,Diffuser Refill", "Diffuser", inplace=True)

In [509]:
if (Perfumes_df.Product_Type == "Arabian").any():
    Perfumes_df.Product_Type.replace("Arabian", "Room Spray", inplace=True)

In [510]:
if (Perfumes_df.Product_Type == "Victoria's Secret").any():
    Perfumes_df.Product_Type.replace("Victoria's Secret", "Perfume Set", inplace=True)

In [511]:
if (Perfumes_df.Product_Type == "Diffuser Refill").any():
    Perfumes_df.Product_Type.replace("Diffuser Refill", "Diffuser", inplace=True)

In [512]:
if (Perfumes_df.Product_Type == "Perfume Set,Hair Mist").any():
    Perfumes_df.Product_Type.replace("Perfume Set,Hair Mist", "Perfume Set", inplace=True)

In [513]:
if (Perfumes_df.Product_Type == "Green Notes, Saffron, Rose, Orris, Sandalwood, Cedar, Balsam, Musk").any():
    Perfumes_df.Product_Type.replace("Green Notes, Saffron, Rose, Orris, Sandalwood, Cedar, Balsam, Musk", "Bakhoor", inplace=True)

In [514]:
if (Perfumes_df.Product_Type == "Laurent Mazzone").any():
    Perfumes_df.Product_Type.replace("Laurent Mazzone", "Perfume Set", inplace=True)

In [515]:
if (Perfumes_df.Product_Type == "Feminine").any():
    Perfumes_df.Product_Type.replace("Feminine", "Perfume", inplace=True)

In [516]:
if (Perfumes_df.Product_Type == "Sophisticated").any():
    Perfumes_df.Product_Type.replace("Sophisticated", "Bakhoor", inplace=True)

In [517]:
if (Perfumes_df.Product_Type == "Classical").any():
    Perfumes_df.Product_Type.replace("Classical", "Room Spray", inplace=True)

In [518]:
if (Perfumes_df.Product_Type == "Glamorous").any():
    Perfumes_df.Product_Type.replace("Glamorous", "Oud", inplace=True)

In [519]:
if (Perfumes_df.Product_Type == "Woody").any():
    Perfumes_df.Product_Type.replace("Woody", "Candle", inplace=True)

In [520]:
if (Perfumes_df.Product_Type == "Aromatic").any():
    Perfumes_df.Product_Type.replace("Aromatic", "Diffuser", inplace=True)

In [521]:
if (Perfumes_df.Product_Type == "Pascal Morabito").any():
    Perfumes_df.Product_Type.replace("Pascal Morabito", "Perfume Set", inplace=True)

In [522]:
if (Perfumes_df.Product_Type == "Eau De Parfum").any():
    Perfumes_df.Product_Type.replace("Eau De Parfum", "Perfume", inplace=True)

In [523]:
if (Perfumes_df.Product_Type == "The lute").any():
    Perfumes_df.Product_Type.replace("The lute", "Perfume Set", inplace=True)
    Perfumes_df.Product_Type.replace("Ozone Notes, Passion Fruit, Agarwood, Amber", "The lute", inplace=True)

In [524]:
if (Perfumes_df.Product_Type == "Natural oud, Oriental Oils").any():
    Perfumes_df.Product_Type.replace("Natural oud, Oriental Oils", "Room Spray", inplace=True)

In [525]:
if (Perfumes_df.Product_Type == "Yves Saint Laurent").any():
    Perfumes_df.Product_Type.replace("Yves Saint Laurent", "Perfume Set", inplace=True)

In [526]:
if (Perfumes_df.Product_Type == "Chopard").any():
    Perfumes_df.Product_Type.replace("Chopard", "Perfume Set", inplace=True)

In [527]:
if (Perfumes_df.Product_Type == "Sensual").any():
    Perfumes_df.Product_Type.replace("Sensual", "Diffuser", inplace=True)

In [528]:
if (Perfumes_df.Product_Type == "Sea scents, jasmine, amber").any():
    Perfumes_df.Product_Type.replace("Sea scents, jasmine, amber", "Room Spray", inplace=True)
    Perfumes_df.Ingredients.replace("Ozone Notes, Passion Fruit, Agarwood, Amber", "Sea scents, jasmine, amber", inplace=True)

In [529]:
if (Perfumes_df.Product_Type == "Soft Floral").any():
    Perfumes_df.Product_Type.replace("Soft Floral", "Candle", inplace=True)
    Perfumes_df.Ingredients.replace("Sea scents, jasmine, amber", "White gardenia, jasmine petals, mongolia flowers, pink nectarine, cotton musk", inplace=True)

In [530]:
if (Perfumes_df.Product_Type == "Lalique").any():
    Perfumes_df.Product_Type.replace("Lalique", "Perfume Set", inplace=True)


In [561]:
if (Perfumes_df.Product_Type == "Romantic").any():
    Perfumes_df.Product_Type.replace("Romantic", "Bakhoor", inplace=True)
    Perfumes_df.Character.replace("Floral", "Romantic", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Roses, fruits, sandalwood, rose, moss, musk", "Floral", inplace=True)
    Perfumes_df.Size.replace("Roses, fruits", "50 g", inplace=True)
    Perfumes_df.Year.replace("Sandalwood, rose", "NaN", inplace=True)
    Perfumes_df.Ingredients.replace("Moss, musk", "Roses, fruits, sandalwood, rose, moss, musk", inplace=True)
    Perfumes_df.Top_note.replace("NaN", "Roses, fruits", inplace=True)
    Perfumes_df.Middle_note.replace("NaN", "Sandalwood, rose", inplace=True)
    Perfumes_df.Base_note.replace("NaN", "Moss, musk", inplace=True)


In [567]:
if (Perfumes_df.Product_Type == "Home Fragrance Set").any():
    Perfumes_df.Brand.replace("250 ml", "Esteban", inplace=True)
    Perfumes_df.Gender.replace("Esteban", "Home", inplace=True)
    Perfumes_df.Fragrance_Family.replace("Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine, Patchouli, Musk", "Aromatic", inplace=True)
    Perfumes_df.Character.replace("Aromatic", "NaN", inplace=True)
    Perfumes_df.Size.replace("500 ml", "250 ml", inplace=True)
    Perfumes_df.Ingredients.replace("Pineapple, papaya, mandarin orange, sugar cane and peach, musk, sandalwood and coconut","Blackcurrant, Bergamot, Neroli, White Geranium, Jasmine, Patchouli, Musk",  inplace=True)
    Perfumes_df.Top_note.replace("Roses, fruits","Blackcurrant, Bergamot", inplace=True)
    Perfumes_df.Middle_note.replace("Sandalwood, rose", "Neroli, White Geranium, Jasmine", inplace=True)
    Perfumes_df.Base_note.replace("Moss, musk","Patchouli, Musk" , inplace=True)

In [546]:
# replacing NaN value with Perfume Set if the name contain Set  
Perfumes_df['Product_Type'].mask(Perfumes_df['Name'].str.contains("Set"),"Perfume Set", inplace=True)


In [540]:
Perfumes_df['Product_Type'].mask(Perfumes_df['Name'].str.contains("Al Majed For Oud"),"Bakhoor", inplace=True)

In [542]:
Perfumes_df['Product_Type'].mask(Perfumes_df['Name'].str.contains("Room Spray"),"Room Spray", inplace=True)

In [543]:
Perfumes_df['Product_Type'].mask(Perfumes_df['Name'].str.contains("Diffuser"),"Diffuser", inplace=True)

In [544]:
Perfumes_df['Product_Type'].mask(Perfumes_df['Name'].str.contains("Candle"),"Candle", inplace=True)

In [549]:
Perfumes_df['Product_Type'].mask(Perfumes_df['Name'].str.contains("Roze"),"Clothes Fragrance", inplace=True)

In [555]:
Perfumes_df['Product_Type'].mask(Perfumes_df['Name'].str.contains("Shower Gel 100 ml"),"Perfume Set", inplace=True)

In [533]:
print(f'The left product type :{Perfumes_df.Product_Type.nunique()}')

The left product type :17


As Shown in the plot Perfumes have the majority as expected  

In [712]:
Perfumes_df.shape

(6565, 18)

#### Cleaning Fragrance Family 

- Fragrance Family Column 

In [534]:
Perfumes_df.Fragrance_Family.value_counts()

Floral                                                                         1346
Oriental                                                                        785
Woody                                                                           641
Aromatic                                                                        415
Fruity                                                                          301
                                                                               ... 
Lemon, Mandarin, Bergamot, Neroli, Mint, Galbanum, Patchouli, Vanilla, Musk       1
Lavender, musk, orange, rose, patchouli, sandalwood, vanilla                      1
Jasmine, lily, musk                                                               1
Floral,Citrus                                                                     1
Verbana, Lemon                                                                    1
Name: Fragrance_Family, Length: 189, dtype: int64

In [535]:
Perfumes_df[Perfumes_df.Fragrance_Family == "Verbana, Lemon"]

Name  Price  \
8786  Aura Myst Essential Oil Verbana - 15 ml   52.0   

                                            Description  Rate Rating_count  \
8786  An intense perfume oil with a fresh scent that...  none         none   

     image  Brand     Gender Product_Type    Character Fragrance_Family  \
8786  none  15 ml  Aura Myst  Perfume Oil  Charismatic   Verbana, Lemon   

        Size Year                                        Ingredients  \
8786  500 ml  NaN  Pineapple, papaya, mandarin orange, sugar cane...   

     Concentration Top_note Middle_note Base_note  
8786           NaN      NaN         NaN       NaN

In [536]:
if (Perfumes_df.Fragrance_Family == "Lemon, Mandarin, Bergamot, Neroli, Mint, Galbanum, Patchouli, Vanilla, Musk").any():
    Perfumes_df.Fragrance_Family.replace("Lemon, Mandarin, Bergamot, Neroli, Mint, Galbanum, Patchouli, Vanilla, Musk", "Floral" , inplace=True)
    Perfumes_df.Ingredients.replace("Oriental notes", "Lemon, Mandarin, Bergamot, Neroli, Mint, Galbanum, Patchouli, Vanilla, Musk", inplace=True)
    Perfumes_df.Size.replace("Eau de Parfum","150 ml" , inplace=True)
    Perfumes_df.Character.replace("150 ml","NaN", inplace=True)
    Perfumes_df.Concentration.replace("NaN","Eau de Parfum", inplace=True)

- Models I probably going with:
2. Cosine Similarity 
3. Deep Learning - Neural Network 
4. content-based similarity model >> go with this 